### PR TITLE
Add new GetRollupData API to enclave client

### PR DESCRIPTION
**Pull request by badgersrus**
### Why this change is needed

We need to be able to retrieve some of the internal rollup data from the enclave when storing data in the host DB.

### What changes were made as part of this PR

* Add new GetRollupData that passes the internal `CalldataRollupHeader` byte slice as an argument and returns the `CalldataRollupHeader.FirstBatchSequence` and `CalldataRollupHeader.StartTime`

### Testing

Can't see a simple way to unit test this so I just manually called the function `processL1BlockTransactions` and ran the full network simulation test and verified the data through console logs. 

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: go/common/enclave.go [hunk 1]
```diff
@@ -109,9 +109,12 @@ type Enclave interface {
 	// GetBatch - retrieve a batch if existing within the enclave db.
 	GetBatch(hash L2BatchHash) (*ExtBatch, SystemError)
 
-	// GetBatchBySeqNo - retrieve batch by sequencer number if it's in the db
+	// GetBatchBySeqNo - retrieve batch by sequencer number if it's in the db.
 	GetBatchBySeqNo(seqNo uint64) (*ExtBatch, SystemError)
 
+	// GetRollupData - retrieve the first batch sequence and start time for a given rollup.
+	GetRollupData(internalRollup []byte) (*big.Int, *uint64, SystemError)
```
Comment (tudor-malene) [inline go/common/enclave.go:116]: I think it's better we return a "PublicRollupMetadata" struct (and a SystemError)

File: go/enclave/enclave.go [hunk 2]
```diff
@@ -295,6 +296,15 @@ func (e *enclaveImpl) GetBatchBySeqNo(seqNo uint64) (*common.ExtBatch, common.Sy
 	return b, nil
 }
 
+func (e *enclaveImpl) GetRollupData(internalRollup []byte) (*big.Int, *uint64, common.SystemError) {
+	calldataRollupHeader := new(common.CalldataRollupHeader)
```
Comment (tudor-malene) [inline go/enclave/enclave.go:300]: this is perfect stuff to retrieve from the database.
They should already be written in the "rollup" table.
It's a matter of selecting them from there

**Review by tudor-malene — CHANGES_REQUESTED** [review]
Thanks Will.
The approach is good. 
Left a few comments

File: go/enclave/enclave.go (hunk 2 above)
Comment (badgersrus) [reply go/enclave/enclave.go]: Done

File: go/common/headers.go [hunk 3]
```diff
@@ -162,6 +162,24 @@ type CalldataRollupHeader struct {
 	ReOrgs [][]byte `rlp:"optional"` // sparse list of reorged headers - non null only for reorgs.
 }
 
+// PublicRollupMetadata contains internal rollup data that can be requested from the enclave.
+type PublicRollupMetadata struct {
```
Comment (tudor-malene) [inline go/common/headers.go:166]: this is one way to marshal the metadata object.
The other way (which is what we've mostly done)  is to use the "converters" in the grpc classes).

You can have a look at `CrossChainMsg` , as an example

**Review by tudor-malene — CHANGES_REQUESTED** [review]
logic looks good.
One small change to the grpc conversion

File: go/enclave/storage/enclavedb/block.go [hunk 4]
```diff
@@ -76,10 +76,6 @@ func updateCanonicalValue(dbtx DBTransaction, isCanonical bool, values []common.
 	dbtx.ExecuteSQL(updateBatches, args...)
 }
 
-func FetchBlockHeader(db *sql.DB, hash common.L1BlockHash) (*types.Header, error) {
```
Comment (tudor-malene) [inline go/enclave/storage/enclavedb/block.go:79]: where has this gone?

**Review by tudor-malene — APPROVED** [review]
lgtm

(one question)

File: go/enclave/storage/enclavedb/block.go (hunk 4 above)
Comment (badgersrus) [reply go/enclave/storage/enclavedb/block.go:79]: It was unused so just got rid of it

**Pull request merged**